pkg/xds/envoy/tags/match_test.go

```diff
@@ -0,0 +1,166 @@
+package tags_test
+
+import (
+	"regexp"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	. "github.com/Kong/kuma/pkg/xds/envoy/tags"
+)
+
+var _ = Describe("MatchingRegex", func() {
+
+	type testCase struct {
+		serviceTags mesh_proto.MultiValueTagSet
+		selector    mesh_proto.SingleValueTagSet
+		expected    bool
+	}
+
+	DescribeTable("should generate regex for matching service's tags",
+		func(given testCase) {
+			// when
+			regexStr := MatchingRegex(given.selector)
+			re, err := regexp.Compile(regexStr)
+			// then
+			Expect(err).ToNot(HaveOccurred())
+			// when
+			matched := re.MatchString(Serialize(given.serviceTags))
+			// then
+			Expect(matched).To(Equal(given.expected))
+		},
+		Entry("match 2 one value tags", testCase{
+			serviceTags: mesh_proto.MultiValueTagSet{
+				"tag1": {"value1": true},
+				"tag2": {"value2": true},
+			},
+			selector: mesh_proto.SingleValueTagSet{
+				"tag1": "value1",
+				"tag2": "value2",
+			},
+			expected: true,
+		}),
+		Entry("match without middle tag2", testCase{
+			serviceTags: mesh_proto.MultiValueTagSet{
+				"tag1": {"value1": true, "value2": true},
+				"tag2": {"value2": true, "value3": true},
+				"tag3": {"value3": true, "value4": true},
+			},
+			selector: mesh_proto.SingleValueTagSet{
+				"tag1": "value1",
```

Comment:
can we change this to `value2` just to see if we match later tag from MultiValuteTagSet?

Reply:
I can't see the change

Reply:
I added a new test case lower:
```
Entry("match the latter valuer", testCase{
	serviceTags: mesh_proto.MultiValueTagSet{
		"tag1": {"value1": true, "value2": true},
	},
	selector: mesh_proto.SingleValueTagSet{
		"tag1": "value2",
	},
	expected: true,
}),
```